ipn/ipnlocal/serve.go

```diff
@@ -91,7 +92,36 @@ func (s *serveListener) Close() error {
 // Listen is retried until the context is canceled.
 func (s *serveListener) Run() {
 	for {
-		ln, err := net.Listen("tcp", s.ap.String())
+		ip := s.ap.Addr()
+		ipStr := ip.String()
+
+		var lc net.ListenConfig
+		if initListenConfig != nil {
+			// On iOS/macOS, this sets the lc.Control hook to
+			// setsockopt the interface index to bind to, to get
+			// out of the network sandbox.
+			if err := initListenConfig(&lc, ip, s.b.prevIfState, s.b.dialer.TUNName()); err != nil {
+				s.logf("serve failed to init listen config %v, backing off: %v", s.ap, err)
+				s.bo.BackOff(s.ctx, err)
+				continue
+			}
+			// On macOS and if we're binding to a privileged port,
+			if runtime.GOOS == "darwin" && s.ap.Port() < 1024 {
```

Comment:
there are three mac variants. you only want to do this for 2 of the 3 probably. see the IsSandboxed funcs in the version package.

Reply:
Done. Using `version.IsSandboxedMacOS()` which is the 2 of the 3 (AppStore and macsys).